core-metadata/src/main/java/org/apache/kylin/metadata/model/Segments.java

```diff
@@ -189,6 +195,21 @@ public void removeLatestSegmentByVolatileRange(Segments<T> segs, long volatileRa
 
     }
 
+    public void removeMaxSpanSegment(Segments<T> segs, long maxSegSpan) {
+        if (maxSegSpan <= 0) {
+            return;
+        }
+        Segments maxSpanSegs = new Segments();
+        for (T seg : segs) {
```

Comment:
Is there any possibility of inner segment(s) being skipped? 
For instance, there are 3 segments like
20190101-20190102, 20190102-20190210, 20190110-20190211.
The method will remove 20190102-20190210, and 2 no-adjacent segments try to be merged

Reply:
@nichunen Added a test for it

Reply:
@lshmouse OK